runtime/autoload/fcnotify.vim

```diff
@@ -0,0 +1,46 @@
+if exists('g:loaded_watcher_provider')
+  finish
+endif
+
+let s:save_cpo = &cpo " save user coptions
+set cpo&vim " reset them to vim defaults
+
+command! -nargs=1 Watch call v:lua.vim.fcnotify.start_watch(expand('<args>'))
+command! -nargs=1 Stop call v:lua.vim.fcnotify.start_watch(expand('<args>'))
+
+" function to prompt the user for a reload
+function! fcnotify#PromptReload(buf)
+  let choice = confirm("File ".bufname(a:buf)." changed. Would you like to reload?","&Yes\n&Show diff\n&No", 1)
+  if choice == 1
+    call fcnotify#Reload(a:buf)
+  elseif choice == 2
+    call fcnotify#DiffOrig(a:buf)
```

Comment:
what happens after showing the diff, I should be able to choose to reload or not.

Reply:
I thought the user can use the `diff` tools to resolve the issue. He could either pull everything or pull selectively and write to the file then. Does this work? Or should there be a prompt after showing the diff too?

Reply:
Ok I don't want to add too much complexity for now. Just looking at ways to make it more user-friendly.
For instance recover.vim detects when there are no changes and it also explicitly differentiates between the on-disk file and inmemory buffer https://github.com/chrisbra/Recover.vim/blob/master/autoload/recover.vim#L299